### PR TITLE
[SHELL32][CONTROL][NCPA] Add file browser fallback

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purpose
Fix some Control Panel functionalities not working when explorer.exe is not running.

JIRA issue: [CORE-19648](https://jira.reactos.org/browse/CORE-19648)

## Proposed changes
If Explorer shell is not available, use ReactOS's alternative file browser instead.

### CODE REVIEW

File: dll/cpl/ncpa/ncpa.c [hunk 1]
```diff
@@ -20,7 +20,10 @@ DisplayApplet(VOID)
 {
 	WCHAR szParameters[] = L"/n,::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{7007ACC7-3202-11D1-AAD2-00805FC1270E}";
 
-	return (INT_PTR) ShellExecuteW(NULL, L"open", L"explorer.exe", szParameters, NULL, SW_SHOWDEFAULT) > 32;
+	/* NOTE: If Explorer shell is not available, use ReactOS's alternative file browser instead. */
```
Comment (HBelusca) [inline dll/cpl/ncpa/ncpa.c:23]:
```suggestion
	/* NOTE: If Explorer shell is not available, use ReactOS's alternative file browser instead */
```

File: dll/win32/shell32/wine/control.c [hunk 2]
```diff
@@ -732,9 +732,10 @@ static	void	Control_DoWindow(CPanel* panel, HWND hWnd, HINSTANCE hInst)
 #else
 static	void	Control_DoWindow(CPanel* panel, HWND hWnd, HINSTANCE hInst)
 {
+    /* NOTE: If Explorer shell is not available, use ReactOS's alternative file browser instead. */
```
Comment (HBelusca) [inline dll/win32/shell32/wine/control.c:735]:
```suggestion
    /* NOTE: If Explorer shell is not available, use ReactOS's alternative file browser instead */
```

**Review by HBelusca — COMMENTED** [review]
In which scenarii would you be using these control panel extensions while explorer is absent?

Which makes me think, whether or not these shellexecute calls that explicitly use "explorer.exe" as the command, aren't hacks that would have to be replaced by some shell32 function COM call that would just take the GUID parameters into account.... :thinking:

**Comment by TAN-Gaming** [comment]
> In which scenarii would you be using these control panel extensions while explorer is absent?

You can open the control panel folder via filebrowser.exe and it works just fine.

**Comment by TAN-Gaming** [comment]
I have the ideas to implement "minimal shell" for ReactOS:
- Rely on starting explorer.exe with commands just like Windows. Add an option in Registry for Explorer to work in filebrowser-only mode, in a less buggy way.
- Use ReactOS's filebrowser.exe as a fallback when Explorer is not available (for applications). Replace hardcoded "explorer.exe" commands in shell32 with COM calls.

**Review by HBelusca — APPROVED** [review]
Let's go with this for the time being; any of the suggestions for further improvements could be implemented in the future.

**Pull request merged**